### PR TITLE
Feature - 1066 Service Actions dialogs

### PR DESCRIPTION
**Pull request by aldipower**
This implements the dialogs/confirm-boxes for the service actions (scale, suspend, destroy).
There is no real action behind them, this will follow next.

Destroy:
confirm-destroy

Scale:
confirm-scale

Suspend:
confirm-suspend

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @orlandohohmeier

**Comment by pierlo-upitup** [comment]
Shouldn't the heading color always match the severity level? @leemunroe

**Comment by pierlo-upitup** [comment]
retest this please

**Comment by aldipower** [comment]
@pierlo-upitup @leemunroe Just FYI - Unfortunately the color of `text-warning` (ugly yellow) compared to `button-warning` differs a lot in Canvas UI. So we don't have matching colors in the moment.

File: src/js/components/ServiceDetail.js [hunk 1]
```diff
@@ -8,11 +9,17 @@ import ServiceDetailConfigurationTab from './ServiceDetailConfigurationTab';
...
 import ServicesBreadcrumb from './ServicesBreadcrumb';
 import TabsMixin from '../mixins/TabsMixin';
 
 const METHODS_TO_BIND = [
   'onActionsItemSelection',
+  'onAcceptDestroyConfirmDialog',
+  'onAcceptSuspendConfirmDialog',
+  'onCancelDestroyConfirmDialog',
+  'onCancelSuspendConfirmDialog',
+  'onCloseScaleFormModal',
```
Comment (pierlo-upitup) [inline src/js/components/ServiceDetail.js]: shouldnt this be `onCancelScaleFormModal`?

File: src/js/components/ServiceDetail.js [hunk 2]
```diff
@@ -30,7 +37,10 @@ class ServiceDetail extends mixin(InternalStorageMixin, TabsMixin) {
 
     this.state = {
       currentTab: Object.keys(this.tabs_tabs).shift(),
-      isServiceFormModalShown: false
+      isServiceFormModalShown: false,
+      isServiceDestroyConfirmShown: false,
+      isServiceScaleFormModalShown: false,
+      isServiceSuspendConfirmShown: false
```
Comment (pierlo-upitup) [inline src/js/components/ServiceDetail.js]: 😱

**Comment by pierlo-upitup** [comment]
Doesn't this PR only implement `ServiceScaleFormModal` ?

**Comment by leemunroe** [comment]
Very nice.

* I'm a bit concerned about the color of the scale and suspend buttons, I would expect the scale button the be standard modal call to action color (green)
* Can we change the width of the instance input? Likely never going to be more than 3 characters so I would expect the width to better reflect that

**Comment by aldipower** [comment]
@pierlo-upitup No, the other dialogs are `Confirm`-elements from reactjs-components as you can see in the code.

@leemunroe Yeah, I will make the buttons to have the standard color and change the width of the input.

File: src/js/components/ServiceDetail.js (hunk 2 above)
Comment (aldipower) [inline src/js/components/ServiceDetail.js]: Yes, really bad.

File: src/js/components/ServiceDetail.js (hunk 1 above)
Comment (aldipower) [inline src/js/components/ServiceDetail.js]: I think the wording for closing a _Modal_ is `close`, we use it like that also at other places.
On the other hand the wording in the _Confirm_ is `cancel`.
So if I change this, the things will be mixed up at another place.
I would like to stick with this.

File: src/js/components/ServiceDetail.js (hunk 2 above)
Comment (orlandohohmeier) [inline src/js/components/ServiceDetail.js]: There is always only one modal open, right? How about one variable holding a modal id or nothing to toggle all these modal states?

Comment (aldipower) [inline src/js/components/ServiceDetail.js]: Good idea, at least this looks much better then. Underlaying problem will remain the same..
Will do it.

**Comment by aldipower** [comment]
@orlandohohmeier I've refactored the show states. Feels much better now.

@leemunroe I've changed the button colors to the `primary`-color as used application wide.
Reducing the width of the input field looks silly though. If you like you could show me what you have exactly in mind?

confirm-scale

Notice the funny text.. ;)
confirm-suspend

**Comment by orlandohohmeier** [comment]
L:eyes:ks good to me! Let's :shipit:

**Comment by Poltergeist** [comment]
Merge

**Comment by kennyt** [comment]
did this PR get merged in with failing integration tests?

**Comment by Poltergeist** [comment]
@kennyt This was my bad, there was one test failing in the integration tests. In the last couple of days the integration tests were failing a lot, not because they were actually failing but because they were flaky that was why I probably didn't paid enough attention to them. I am sorry for this and will pay more attention next time.

**Pull request merged**